### PR TITLE
Accept top level statements

### LINKED ISSUE #168: Inject PHP code without PHP tags
Hello!

I'm trying (somewhat unsuccessfully, but trying nonetheless) to write a parser for the [Laravel Blade](https://laravel.com/docs/master/blade) templating system. I've captured the code content between the output directives (e.g., `{{ $fooBar->some_attribute }}`) but when I inject the content the parser thinks it (`$fooBar->some_attribute`) is just text.

If I do the following then the parser recognizes it as a variable and properly highlights it, however, this defeats the whole purpose of the templating system:
`{{ <?php $fooBar->some_attribute ?> }}`

Something similar occurs in markdown file with PHP fenced content:

E.g., here the php parser thinks the content is just text
image

whereas it's a variable assignment with the php tags
image

Is there a way to have the parser recognize the content as php without explicitly declaring the php tags?

Thanks!

### PR DESCRIPTION
**Pull request by CalebDW**
Closes #168 

Checklist:
- [x] All tests pass in CI
- [x] There are sufficient tests for the new fix/feature
- [x] Grammar rules have not been renamed unless absolutely necessary (x rules renamed)
- [x] The conflicts section hasn't grown too much (x new conflicts)
- [x] The parser size hasn't grown too much (master: STATE_COUNT, PR: STATE_COUNT) (check the value of STATE_COUNT in src/parser.c)

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -103,7 +103,10 @@ module.exports = grammar({
       optional(seq(
         $.php_tag,
         repeat($._statement)
-      ))
+      )),
+      // accept statements at top-level to enable
+      // parsing of partial code snippets (e.g, in markdown; see #168).
+      optional(repeat($._statement)),
```
Comment (aryx) [inline grammar.js:109]: I'm not sure the statement will actually be parsed because of the optional($.text) at the top that may "eat" every tokens.

**Review by aryx — CHANGES_REQUESTED** [review]
Please add some tests in corpus/ and regenerate the parser.c and other files

File: grammar.js (hunk 1 above)
Comment (CalebDW) [reply grammar.js]: Hmmm...you're right, I tried moving this line to the top of the `program` but the text still swallows everything up (see failing test)

Comment (CalebDW) [reply grammar.js]: This is more complicated than I originally thought....do you have any idea on how to allow top-level statement parsing for injected code snippets while still allowing HTML?

**Comment by CalebDW** [comment]
Closing as accepting top level statements is not possible with this parser in its current state: the `text` is too hungry.

As mentioned in the above comment, the solution is probably to create two separate parsers in this repo (similar to the [markdown parser](https://github.com/MDeiml/tree-sitter-markdown)), one which is `php-base` or `php-only` and the other (`php`) which inject the `php-base` into the php tag nodes

**Pull request closed without merging**